apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/pagerules/PageRuleSteps.java

```diff
@@ -0,0 +1,175 @@
+package gov.cdc.nbs.questionbank.pagerules;
+
+import gov.cdc.nbs.questionbank.model.CreateRuleRequest;
+import gov.cdc.nbs.questionbank.model.ViewRuleResponse;
+import gov.cdc.nbs.questionbank.pagerules.response.CreateRuleResponse;
+import gov.cdc.nbs.questionbank.support.ExceptionHolder;
+import gov.cdc.nbs.questionbank.support.UserMother;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+
+public class PageRuleSteps {
+
+    @Autowired
+    private PageRuleController pageRuleController;
+
+    @Autowired
+    private ExceptionHolder exceptionHolder;
+
+    CreateRuleRequest ruleRequest;
+    CreateRuleResponse ruleResponse;
+
+    Page<ViewRuleResponse> viewRuleResponses;
+
+    @Autowired
+    private UserMother userMother;
+    Long ruleId;
+
+    @Given("I am not logged in user")
+    public void i_am_not_logged_in() {
+        SecurityContextHolder.getContext().setAuthentication(null);
+    }
+
+    @When("I make a request to add a rule to a page")
+    public void i_make_a_request_to_add_a_rule_to_a_page() {
+
+        try {
+            ruleResponse = pageRuleController.createBusinessRule(ruleRequest());
+        } catch (AccessDeniedException e) {
+            exceptionHolder.setException(e);
+        } catch (AuthenticationCredentialsNotFoundException e) {
+            exceptionHolder.setException(e);
+        }
+    }
+    @Then("A rule is created")
+    public void a_rule_is_created() {
+        assertNotNull(ruleResponse);
+    }
+
+    @Then("A no credentials found exception is thrown")
+    public void a_no_credentials_found_exception_is_thrown() {
+        assertNotNull(exceptionHolder.getException());
+        assertTrue(exceptionHolder.getException() instanceof AuthenticationCredentialsNotFoundException);
+    }
+
+
+    @When("I make a request to update a rule to a page")
+    public void i_make_a_request_to_update_a_rule_to_a_page() {
+        try{
+            ruleId = 6405L;
+            ruleResponse = pageRuleController.updatePageRule(ruleId,ruleRequest());
+        } catch (AuthenticationCredentialsNotFoundException e) {
+            exceptionHolder.setException(e);
+        }
+    }
+
+    @Then("A rule is updated")
+    public void a_rule_is_updated() {
+        assertNotNull(ruleResponse);
+    }
```

Comment:
Would be good to verify the contents of the response as well. Making sure that the response matches the values that were passed in the request